qiskit_machine_learning/neural_networks/opflow_qnn.py

```diff
@@ -0,0 +1,136 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An Opflow Quantum Neural Network that allows to use a parametrized opflow object as a
+neural network."""
+
+from copy import deepcopy
+from typing import List, Optional, Union, Tuple, Dict
+
+import numpy as np
+from qiskit.circuit import Parameter
+from qiskit.opflow import Gradient, CircuitSampler, ListOp, OperatorBase, ExpectationBase
+from qiskit.providers import BaseBackend, Backend
+from qiskit.utils import QuantumInstance
+from qiskit.utils.backend_utils import is_aer_provider
+
+from .neural_network import NeuralNetwork
+from .. import QiskitMachineLearningError
+
+
+class OpflowQNN(NeuralNetwork):
+    """Opflow Quantum Neural Network."""
+
+    def __init__(self, operator: OperatorBase,
+                 input_params: Optional[List[Parameter]] = None,
+                 weight_params: Optional[List[Parameter]] = None,
+                 exp_val: Optional[ExpectationBase] = None,
+                 gradient: Optional[Gradient] = None,
+                 quantum_instance: Optional[Union[QuantumInstance, BaseBackend, Backend]] = None):
+        """Initializes the Opflow Quantum Neural Network.
+
+        Args:
+            operator: The parametrized operator that represents the neural network.
```

Comment:
This does not have parameterized in () - but given the following params default to None I guess though this is analgous to the circuit based ine

Reply:
Same as for the circuits.